src/charm.py

```diff
@@ -138,38 +154,60 @@ def _publish_nrf_info_for_all_requirers(self, url: str) -> None:
             return
         self.nrf_provider.set_nrf_information_in_all_relations(url)
 
-    @property
-    def _config_file_is_written(self) -> bool:
-        """Returns whether the config file was written to the workload container.
+    def _relation_created(self, relation_name: str) -> bool:
+        """Returns whether a given Juju relation was crated.
+
+        Args:
+            relation_name (str): Relation name
 
         Returns:
-            bool: Whether the config file was written.
+            bool: Whether the relation was created.
         """
-        if not self._container.exists(f"{BASE_CONFIG_PATH}/{CONFIG_FILE_NAME}"):
-            logger.info(f"Config file is not written: {CONFIG_FILE_NAME}")
-            return False
-        logger.info("Config file is written")
-        return True
+        return bool(self.model.get_relation(relation_name))
 
-    @property
-    def _database_relation_is_created(self) -> bool:
-        """Returns whether database relation is created.
+    @staticmethod
+    def _render_config(
+        database_name: str,
+        database_url: str,
+        nrf_sbi_port: int,
+    ) -> str:
+        jinja2_environment = Environment(loader=FileSystemLoader("src/templates/"))
+        template = jinja2_environment.get_template("nrfcfg.yaml.j2")
+        content = template.render(
+            database_name=database_name,
+            database_url=database_url,
+            nrf_sbi_port=nrf_sbi_port,
+        )
+        return content
```

Comment:
Please move this out of the class and remove `@staticmethod`

Reply:
it was discussed in a different pr, that this function is really related to the class (logically at least) so it makes sense to keep it there. However personally I don't have a strong opinion about the issue.
I moved it out of the class. ✅